lib/ansible/modules/windows/win_domain_group.ps1

```diff
@@ -0,0 +1,354 @@
+    $credential = New-Object -TypeName System.Management.Automation.PSCredential -ArgumentList $username, $domain_password
+    $extra_args.Credential = $credential
+}
+
+try {
+    $group = Get-ADGroup -Identity $name -Properties * @extra_args
+} catch [Microsoft.ActiveDirectory.Management.ADIdentityNotFoundException] {
+    $group = $null
+} catch {
+    Fail-Json $result "failed to retrieve initial details for group $($name): $($_.Exception.Message)"
+}
+if ($state -eq "absent") {
+    if ($group -ne $null) {
+        if ($group.ProtectedFromAccidentalDeletion -eq $true -and $force -eq $true) {
+            $group = $group | Set-ADObject -ProtectedFromAccidentalDeletion $false -WhatIf:$check_mode -PassThru @extra_args
+        } elseif ($group.ProtectedFromAccidentalDeletion -eq $true -and $force -eq $false) {
+            Fail-Json $result "cannot delete group $name when ProtectedFromAccidentalDeletion is turned on, run this module with force=true to override this"
+        }
+
+        try {
+            $group | Remove-ADGroup -Confirm:$false -WhatIf:$check_mode @extra_args
+        } catch {
+            Fail-Json $result "failed to remove group $($name): $($_.Exception.Message)"
+        }
+        
+        $result.changed = $true
+        if ($diff_mode) {
+            $result.diff.prepared = "-[$name]"
+        }
+    }
+} else {
+    # validate that path is an actual path
+    if ($path -ne $null) {
+        try {
+            Get-ADObject -Identity $path @extra_args | Out-Null
+        } catch [Microsoft.ActiveDirectory.Management.ADIdentityNotFoundException] {
+            Fail-Json $result "the group path $path does not exist, please specify a valid LDAP path"
+        }
+    }
+
+    $diff_text = $null
+    if ($group -ne $null) {
+        # will be overriden later if no change actually occurs
+        $diff_text += "[$name]`n"
+
+        # change the path of the group
+        if ($path -ne $null) {
+            $group_cn = $group.CN
+            $existing_path = $group.DistinguishedName -replace "^CN=$group_cn,",''
+            if ($existing_path -ne $path) {
+                $protection_disabled = $false
+                if ($group.ProtectedFromAccidentalDeletion -eq $true -and $force -eq $true) {
+                    $group | Set-ADObject -ProtectedFromAccidentalDeletion $false -WhatIf:$check_mode -PassThru @extra_args | Out-Null
+                    $protection_disabled = $true
+                } elseif ($group.ProtectedFromAccidentalDeletion -eq $true -and $force -eq $false) {
+                    Fail-Json $result "cannot move group $name when ProtectedFromAccidentalDeletion is turned on, run this module with force=true to override this"
+                }
+
+                try {
+                    $group = $group | Move-ADObject -Targetpath $path -WhatIf:$check_mode -PassThru @extra_args
+                } catch {
+                    if ($protection_disabled -eq $true) {
```

Comment:
Maybe put this in an enclosing `try/finally` to eliminate the need to repeat it below?